Index: CodeSniffer/Standards/Generic/Sniffs/Files/LineLengthSniff.php
===================================================================
--- CodeSniffer/Standards/Generic/Sniffs/Files/LineLengthSniff.php	(revision 303234)
+++ CodeSniffer/Standards/Generic/Sniffs/Files/LineLengthSniff.php	(working copy)
@@ -49,6 +49,13 @@
      */
     public $absoluteLineLimit = 100;
 
+    /**
+     * Whether to use mb_strlen for measuring line length
+     *
+     * @var bool
+     */
+    public $useMultibyteString = false;
+
 
     /**
      * Returns an array of tokens this test wants to listen for.
@@ -118,7 +125,7 @@
         if (preg_match('|@version[^\$]+\$Id|', $lineContent) === 0
             && preg_match('|@license|', $lineContent) === 0
         ) {
-            $lineLength = strlen($lineContent);
+            $lineLength = $this->useMultibyteString ? mb_strlen($lineContent) : strlen($lineContent);
             if ($this->absoluteLineLimit > 0
                 && $lineLength > $this->absoluteLineLimit
             ) {
