Index: tests/Core/File/PassByReference.inc
===================================================================
--- tests/Core/File/PassByReference.inc	(revision 0)
+++ tests/Core/File/PassByReference.inc	(revision 0)
@@ -0,0 +1,7 @@
+<?php
+function passByReference(&$var)
+{
+
+}
+
+?>
Index: tests/Core/File/ArrayHint.inc
===================================================================
--- tests/Core/File/ArrayHint.inc	(revision 0)
+++ tests/Core/File/ArrayHint.inc	(revision 0)
@@ -0,0 +1,7 @@
+<?php
+function arrayHint(array $var)
+{
+
+}
+
+?>
Index: tests/Core/File/GetMethodParametersTest.php
===================================================================
--- tests/Core/File/GetMethodParametersTest.php	(revision 0)
+++ tests/Core/File/GetMethodParametersTest.php	(revision 0)
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Tests for the PHP_CodeSniffer_File class
+ *
+ * PHP version 5
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    Anti Veeranna <duke@masendav.com>
+ * @copyright 2009 SQLI <www.sqli.com>
+ * @copyright 2006 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   http://matrix.squiz.net/developer/tools/php_cs/licence BSD Licence
+ * @version   CVS: $Id: IsCamelCapsTest.php 240585 2007-08-02 00:05:40Z squiz $
+ * @link      http://pear.php.net/package/PHP_CodeSniffer
+ */
+
+require_once 'PHPUnit/Framework/TestCase.php';
+
+/**
+ * Tests for the PHP_CodeSniffer reporting system.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    Anti Veeranna <duke@masendav.com>
+ * @copyright 2009 SQLI <www.sqli.com>
+ * @copyright 2006 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   http://matrix.squiz.net/developer/tools/php_cs/licence BSD Licence
+ * @version   Release: @package_version@
+ * @link      http://pear.php.net/package/PHP_CodeSniffer
+ */
+class Core_File_GetMethodParametersTest extends PHPUnit_Framework_TestCase
+{
+
+    private $_phpcsFile;
+
+    /**
+     * Initialize & tokenize PHP_CodeSniffer_File with sample code for current test
+     *
+     * @return void
+     */
+    public function setUp()
+    {
+        // get name of sample code out of the name of test
+        // testPassByReference -> DIR/PassByReference.inc
+        $file = dirname(__FILE__) . '/' . substr($this->getName(), 4) . '.inc';
+        $contents = file_get_contents($file);
+
+        $Sniffer = new PHP_CodeSniffer();
+
+        $this->_phpcsFile = new PHP_CodeSniffer_File(
+            $file,
+            array(),
+            array(),
+            array(),
+            $Sniffer
+        );
+
+        $this->_phpcsFile->start($contents);
+
+    }//end setUp()
+
+
+    /**
+     * Clean up after finished test
+     *
+     * @return void
+     */
+    public function tearDown()
+    {
+        unset($this->_phpcsFile);
+
+    }//end tearDown()
+
+
+    /**
+     * Verify pass-by-reference parsing
+     *
+     * @return void
+     */
+    public function testPassByReference()
+    {
+        $expected = array();
+        $expected[0] = array(
+            'name' => '$var'
+            , 'pass_by_reference' => true
+            , 'type_hint' => ''
+        );
+
+        $functionPtr = $this->_phpcsFile->findNext(T_FUNCTION, 0);
+        $actual = $this->_phpcsFile->getMethodParameters($functionPtr);
+
+        $this->assertSame($expected, $actual);
+
+    }//end testPassByReference()
+
+
+    /**
+     * Verify array hint parsing
+     *
+     * @return void
+     */
+    public function testArrayHint()
+    {
+
+        $expected = array();
+        $expected[0] = array(
+            'name' => '$var'
+            , 'pass_by_reference' => false
+            , 'type_hint' => 'array'
+        );
+
+        $functionPtr = $this->_phpcsFile->findNext(T_FUNCTION, 0);
+        $actual = $this->_phpcsFile->getMethodParameters($functionPtr);
+
+        $this->assertSame($expected, $actual);
+
+    }//end testArrayHint()
+
+
+}//end class
+
+?>
+
Index: tests/Core/AllTests.php
===================================================================
--- tests/Core/AllTests.php	(revision 302851)
+++ tests/Core/AllTests.php	(working copy)
@@ -26,6 +26,7 @@
 require_once 'Reports/SourceTest.php';
 require_once 'Reports/SvnblameTest.php';
 require_once 'Reports/GitblameTest.php';
+require_once 'File/GetMethodParametersTest.php';
 
 if (is_file(dirname(__FILE__).'/../../CodeSniffer.php') === true) {
     // We are not installed.
@@ -85,6 +86,7 @@
         $suite->addTestSuite('Core_Reports_SourceTest');
         $suite->addTestSuite('Core_Reports_SvnblameTest');
         $suite->addTestSuite('Core_Reports_GitblameTest');
+        $suite->addTestSuite('Core_File_GetMethodParametersTest');
         return $suite;
 
     }//end suite()
